### PR TITLE
Panda, Eagle, and Tiger levels

### PR DESCRIPTION
**Pull request by HerbCSO**
Not really quite sure I have the specs understood correctly for the Tiger level.

### CODE REVIEW

File: movie_json.rb [hunk 1]
```diff
@@ -1,21 +1,55 @@
...
+find_movie(movies)
 
 while true do
-  puts "Search Again (Y/N)" 
-  answer = gets.upcase[0]
-  if answer == "Y"
-    find_movie
+  puts "Add Another (Y/N)?"
+  if gets =~ /Y/i
+    find_movie(movies)
```
Comment (jwo) [inline movie_json.rb:38]: I like that you're sending in the movies here and injecting this dependency.

**Comment by jwo** [comment]
It looks pretty good. Some things I might have done:

1) Not used string interpretation on the movie to tell it was not-found. So you could either have an UnknownMovie class, or a method on the Movie that tells if its valid... `if movie.valid?`
2) You could stub out the Api call to force that the movie is not found -- that way, your tests would stay "fast" and work even when the internet is down.

**Comment by HerbCSO** [comment]
Good points @jwo - I added a Movie.valid? method and have now stubbed out the Api call for the not found methods, included the new fixture file for that.

Is there some way I can add tests for movie_json.rb? I'm having trouble because it's not a class and it seems that the gets calls are causing rspec to fail. From a quick googling it appears that simulating the IO is a little complicated in Rspec, so I'm wondering if it would be worth splitting out the extra functions into their own class for testing or if there's a better way to approach this - advice/guidance appreciated!

Going to the beach for a while now, probably no more updates from me until tomorrow... ;]

**Comment by jwo** [comment]
You're on the right track that movie_json is getting complicated in the Tiger level and should be its own object.

At that point, it's really a "Game" or "App" object, and you would abstract away the gets/puts.  So the App just asks for input and gives output. That gets complicated quickly though, but it's doable and then easier to test. I just didn't for this class to keep it a bit simpler.  

# 

Jesse Wolgamott @ Comal Productions, LLC.
Web. Mobile. Design. Training.
@jwo
http://comalproductions.com

**Pull request closed without merging**